src/System.Console/tests/WindowAndCursorProps.cs

```diff
@@ -33,8 +33,16 @@ public static void BufferSize_GettingSameAsWindowSize()
     [Fact]
     public static void WindowWidth_WindowHeight_InvalidSize()
     {
-        Assert.Throws<ArgumentOutOfRangeException>("width", () => Console.WindowWidth = 0);
-        Assert.Throws<ArgumentOutOfRangeException>("height", () => Console.WindowHeight = 0);
+        if (Console.IsOutputRedirected)
```

Comment:
I see it is the accessing of the other property getter that causes the redirection check to occur. While this behavior was likely not intended I think we can live with it.